algorithms/src/Kokkos_Random.hpp

```diff
@@ -886,29 +886,27 @@ class Random_XorShift64_Pool {
  public:
   using generator_type = Random_XorShift64<DeviceType>;
 
-  KOKKOS_INLINE_FUNCTION
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_FUNCTION
   Random_XorShift64_Pool() {
```

Comment:
unless i am misunderstanding, given how `KOKKOS_ENABLE_DEPRECATED_CODE_4` is guarding this code now, if we remove the depreacted code there is no constructor for the device.

Reply:
PR was updated but we are still intentionally removing the device-side default constructor.